webstruct/sequence_encoding.py

```diff
@@ -158,6 +159,148 @@ def iter_group(cls, data, strict=False):
         if buf:
             yield buf, tag
 
+class BilouEncoder(object):
+    """
+    Utility class for encoding tagged token streams using BILOU encoding.
+
+    Encode input tokens using ``encode`` method::
+
+        >>> bilou_encoder = BilouEncoder()
+        >>> input_tokens = ["__START_PER__", "John", "__END_PER__", "said"]
+        >>> def encode(encoder, tokens): return [p for p in BilouEncoder.from_indices(encoder.encode(tokens), tokens)]
+        >>> encode(bilou_encoder, input_tokens)
+        [('John', 'U-PER'), ('said', 'O')]
+
+
+        >>> input_tokens = ["hello", "__START_PER__", "John", "Doe", "__END_PER__", "__START_PER__", "Mary", "__END_PER__", "said"]
+        >>> tokens = encode(bilou_encoder, input_tokens)
+        >>> tokens, tags = bilou_encoder.split(tokens)
+        >>> tokens, tags
+        (['hello', 'John', 'Doe', 'Mary', 'said'], ['O', 'B-PER', 'L-PER', 'U-PER', 'O'])
+
+    To reset internal state, use ``reset method``::
+
+        >>> bilou_encoder.reset()
+
+    """
+    def __init__(self, token_processor=None):
+        self.token_processor = token_processor or InputTokenProcessor()
+        self.reset()
+
+    def reset(self):
```

Comment:
reset method is pointless, as restarts are not supported

Reply:
reset is needed to clean the encoder state before processing new content https://github.com/scrapinghub/webstruct/blob/7ddd3170d3a5feec20e430abe9fd41619066a71d/webstruct/html_tokenizer.py#L170